beets/ui/commands.py

```diff
@@ -708,6 +736,15 @@ def choose_match(self, task):
                     _, _, candidates, rec = autotag.tag_album(
                         task.items, search_id=search_id
                     )
+            elif choice in extra_ops.keys():
+                # Allow extra ops to automatically set the post-choice.
+                # TODO: currently action.MANUAL and action.MANUAL_ID are not
+                # properly handled
+                post_choice = extra_ops[choice](self, task)
+                if post_choice in (importer.action.SKIP, importer.action.ASIS,
+                                   importer.action.TRACKS,
+                                   importer.action.ALBUMS):
+                    return post_choice
```

Comment:
This is illustrated on `mbsubmit.print_tracks_and_skip()`: if a callback returns a value, the main loop returns that value as well. The idea and main benefit is to save the user an extra prompt - not sure if it introduces more problems/lack of intuitiveness than it solves, as some of the actions (`MANUAL` and `MANUAL_ID`) do not really exit the loop, they are currently not really handled properly.